discord/guild.go

```diff
@@ -141,6 +141,38 @@ func (guild *GuildState) handleTrackedMembers(sm *SessionManager, delay int, han
 	priorityQueue := &PatchPriority{}
 	heap.Init(priorityQueue)
 
+	// amountWorkers value determines how many goroutines will be spawned to mute users
+	// this amount could be increased up to 10, but 8 should be enough
+	const amountWorkers = 8
+
+	// wait group
+	wgWorkers := sync.WaitGroup{}
+	wgWorkers.Add(amountWorkers)
+
+	// buffered channel of size 10 (maximum numbers os players in a match)
+	userChan := make(chan UserPatchParameters, 10)
+
+	// starts workers, aka channel receivers
+	for i := 0; i < amountWorkers; i++ {
+		// Receive values until userChan is
+		// closed and the value buffer queue
+		// of userChan becomes empty.
+		go func() {
+			defer wgWorkers.Done()
+
+			// read messages from the channel and act on them
+			for parameters := range userChan {
+				guildMemberUpdate(sm.GetSessionForRequest(), parameters)
+				log.Println(parameters)
```

Comment:
Just a note: Reading from a channel this way is blocking, so all workers will be asleep when they reach this part.